daemon/cmd/endpoint.go

```diff
@@ -306,26 +306,32 @@ func (m *endpointCreationManager) DebugStatus() (output string) {
 // createEndpoint attempts to create the endpoint corresponding to the change
 // request that was specified.
 func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, epTemplate *models.EndpointChangeRequest) (*endpoint.Endpoint, int, error) {
-	if option.Config.EnableEndpointRoutes {
-		if epTemplate.DatapathConfiguration == nil {
-			epTemplate.DatapathConfiguration = &models.EndpointDatapathConfiguration{}
+	if epTemplate.DatapathConfiguration == nil {
+		dpConfig := endpoint.NewDatapathConfiguration()
+		epTemplate.DatapathConfiguration = &dpConfig
+	} else {
+		if option.Config.EnableEndpointRoutes {
+			// Indicate to insert a per endpoint route instead of routing
+			// via cilium_host interface
+			epTemplate.DatapathConfiguration.InstallEndpointRoute = true
+			// Since routing occurs via endpoint interface directly, BPF
+			// program is needed on that device at egress as BPF program on
+			// cilium_host interface is bypassed
+			epTemplate.DatapathConfiguration.RequireEgressProg = true
+			// Delegate routing to the Linux stack rather than tail-calling
+			// between BPF programs.
+			disabled := false
+			epTemplate.DatapathConfiguration.RequireRouting = &disabled
+		} else {
+			epTemplate.DatapathConfiguration.InstallEndpointRoute = false
+			epTemplate.DatapathConfiguration.RequireEgressProg = false
+			epTemplate.DatapathConfiguration.RequireRouting = nil
 		}
-
-		// Indicate to insert a per endpoint route instead of routing
-		// via cilium_host interface
-		epTemplate.DatapathConfiguration.InstallEndpointRoute = true
-
-		// Since routing occurs via endpoint interface directly, BPF
-		// program is needed on that device at egress as BPF program on
-		// cilium_host interface is bypassed
-		epTemplate.DatapathConfiguration.RequireEgressProg = true
-
-		// Delegate routing to the Linux stack rather than tail-calling
-		// between BPF programs.
-		disabled := false
-		epTemplate.DatapathConfiguration.RequireRouting = &disabled
 	}
 
+	dpConfig := endpoint.NewDatapathConfiguration()
+	epTemplate.DatapathConfiguration = &dpConfig
```

Comment:
This is still overriding the configuration provided by cilium-cni right?

Reply:
:facepalm: that's a leftover from copy/paste while fixing the above. Fixed now. Sorry about that.